packages/base/source/adt/union/union.js

```diff
@@ -82,8 +108,11 @@ instead to check if a value belongs to the ADT variant.`);
        *   where 'b = 'a[`@@folktale:adt:tag]
        */
       matchWith(pattern) {
-        return pattern[name](this);
-      } 
+        assertObject('Union.mapObject#matchWith', pattern);
+        if (name in pattern) { return pattern[name](this); }
+        else if (ANY in pattern) { return pattern[ANY]() }
+        else { throw new Error(getMatchWithErrorMessage(pattern, name)) }
+      }
```

Comment:
This is a minor nit-picking, but could you format this with each statement in its own line?

```js
if (name in pattern) {
  return pattern[name](this);
} else if (ANY in pattern) {
  return pattern[ANY]();
} else {
  throw new Error(getMatchWithErrorMessage(pattern, name));
} 
```